macros/cargo

```diff
@@ -44,5 +56,9 @@ rustflags = %{__global_rustflags_static_toml}\
 EOF\
 )
 
-%cargo_build %{__cargo_cross_env} %{__cargo} build %{__cargo_cross_opts} --release %{?cargo_args}
-%cargo_build_static %{__cargo_cross_env_static} %{__cargo} build %{__cargo_cross_opts_static} --release %{?cargo_args}
+%cargo_build %set_cross_rust_env %{__cargo_cross_env}  %{__cargo} build %{__cargo_cross_opts} --release %{?cargo_args}
```

Comment:
I don't like using `%set_cross_rust_env` here; it means we'll be rewriting config files several times potentially, which is unexpected behavior for a compiler invocation.

We can run both the cmake toolchain macros as part of `%cargo_prep`, and set the `CMAKE_TOOLCHAIN_FILE` and `BINDGEN_EXTRA_CLANG_ARGS` variables to the right values in the `%_cargo_cross_env_*` macros.